scripts/albs-1147.py

```diff
@@ -0,0 +1,231 @@
+            record.title = debranded_title
+            record.description = debranded_description
+            affected_records[record.id] = {
+                'title': record.title,
+                'description': record.description,
+                'content_ptr_id': record.content_ptr_id
+            }
+            record.updated_date = datetime.datetime.utcnow().strftime(
+                "%Y-%m-%d %H:%M:%S",
+            )
+
+        if write:
+            session.commit()
+
+    log.info(f'{os.linesep * 2}Looking for records in almalinux\'s \'errata_records\' table...')
+
+    async with asynccontextmanager(get_db)() as session, session.begin():
+        result = await session.execute(
+            select(ErrataRecord).where(
+                ErrataRecord.id.in_(list(affected_records.keys()))
+            )
+        )
+        records = result.scalars().all()
+        log.info(f'Found {len(records)} records.')
+
+        for record in records:
+            log.info(f'{record.id} - {record.title or record.original_title}')
+            log_differences(record.original_title, affected_records[record.id]['title'])
+            log_differences(record.original_description, affected_records[record.id]['description'])
+            record.original_title = affected_records[record.id]['title']
+            record.original_description = affected_records[record.id]['description']
+            record.updated_date = datetime.datetime.utcnow()
+
+        if write:
+            await session.commit()
+
+    repos_to_publicate = set()
+    pulp = PulpClient(settings.pulp_host, settings.pulp_user, settings.pulp_password)
+
+    with get_pulp_db() as pulp_db, pulp_db.begin():
+        repos = pulp_db.execute(select(CoreRepository)).scalars().all()
+        for repo in repos:
+            repo_href = f"/pulp/api/v3/repositories/rpm/rpm/{repo.pulp_id}/"
+
+            repo_subq = (
+                select(CoreRepository.pulp_id).where(
+                    CoreRepository.name == repo.name
+                )
+            )
+
+            repocontent_subq = (
+                select(CoreRepositoryContent.content_id).where(
+                    CoreRepositoryContent.repository_id.in_(repo_subq)
+                )
+            )
+
+            content_subq = (
+                select(CoreContent.pulp_id).where(
+                    CoreContent.pulp_type == 'rpm.advisory',
+                    CoreContent.pulp_id.in_(repocontent_subq)
+                )
+            )
+
+            updaterecord_subq = (
+                select(UpdateRecord).where(
+                    and_(
+                        UpdateRecord.content_ptr_id.in_(content_subq),
+                        UpdateRecord.content_ptr_id.in_(
+                            {v['content_ptr_id'] for k, v in affected_records.items()})
+                    )
+                )
+            )
+
+            records = pulp_db.execute(updaterecord_subq).scalars().all()
```

Comment:
this approach will trigger a lot of requests, you can simplify it by collecting the affected `UpdateRecord.content_ptr_id` at first and then gather the needed `repository_ids` with one request like this
```python3
affected_errata_ids = []
for record in records:
    ...
    affected_errata_ids.append(record.content_ptr_id)
...
query = (
    select(CoreRepositoryContent.repository_id)
    .where(
        CoreRepositoryContent.version_removed_id.is_(None),
        CoreRepositoryContent.content_id.in_(affected_errata_ids)
    )
)
```